src/mlpack/core/data/load_image.hpp

```diff
@@ -0,0 +1,150 @@
+/**
+ * @file load_image.hpp
+ * @author Mehul Kumar Nirala
+ *
+ * An image loading utillity
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_LOAD_IMAGE_HPP
+#define MLPACK_CORE_DATA_LOAD_IMAGE_HPP
+
+#include <string>
+#include <iostream>
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/log.hpp>
+#include <mlpack/prereqs.hpp>
+
+#ifdef HAS_FILESYSTEM
```

Comment:
We have to set the flag, similar to https://stackoverflow.com/a/54290906.

Reply:
I think, in some cases, it will be `<filesystem>` instead.  You might need to do extra macro checking to see what is available for what compilers (and we should figure out what happens on Windows too because we have lots of Windows users).